packages/fxa-settings/src/lib/auth-errors/auth-errors.ts

```diff
@@ -604,13 +621,34 @@ const ERRORS = {
 };
 
 type ErrorKey = keyof typeof ERRORS;
-type ErrorVal = { errno: number; message: string };
+type ErrorVal = { errno: number; message: string; version?: number };
+
+export const composeAuthUiErrorTranslationID = (errno?: number) => {
+  /* all of these checks for fields/values being present look a little wonky, but allow us to work with
+   * the AuthUiError type, which has all optional fields. Previously this wasn't an issue bc we didn't use
+   * a utility.
+   */
+  if (errno && errno in AuthUiErrorNos) {
+    const error = AuthUiErrorNos[errno];
+    return `auth-error-${errno}${error.version ? ('-' + error.version) : ''}`;
+  }
+  /*
+   * We opt to return an empty string instead of throwing an error so that this can't break in prod.
+   * Instead, we log to sentry.
+   */
+  const message = errno ?
+  `composeAuthUiErrorTranslationID: There is no matching error in AuthUiErrors for errno: ${errno}` :
+  'composeAuthUiErrorTranslationID: No error number given, unable to create a localization ID for AuthUiError string.'
```

Comment:
@millsoper Hmm seeing this second message really makes me think this is going to be better if that entire `error` object is passed in, because if this is logged to Sentry with no additional details, we really can't do anything about it except look at the breadcrumbs. It'd be really nice to be able to report the `message` coming off of the error so we've got something to go off of if this does happen. I know you said that causes a lot of test failures and TS errors, though. 😕 Does passing in `errno` and `message` as two separate params...?

I guess my other question looking at this again is when `errno` is _ever_ undefined. I know we Slacked about that though and I think you mentioned it's because of a hodgepodge of JS/TS and being unable to tell if any of them might be missing so it's safest to set this type as possibly undefined, right?

Reply:
Okay I've been mulling this over and as much as I want this PR to go in...I think you're right and I'm going to try to get the whole error in 😂 I think I almost have it working